specutils/spectra/spectrum1d.py

```diff
@@ -91,17 +92,54 @@ def to_flux(self, unit):
             The converted flux array.
         """
         new_data = self.flux.to(
-            unit, equivalencies=eq.spectral_density(self.dispersion))
+            unit, equivalencies=eq.spectral_density(self.spectral_axis))
 
         self._data = new_data.value
         self._unit = new_data.unit
 
         return self.flux
 
-    def to_dispersion(self, unit, rest, type="doppler_relativistic"):
+    @property
+    def frequency(self):
+        return self._spectral_axis.to(u.GHz, u.spectral())
+
+    @property
+    def wavelength(self):
+        return self._spectral_axis.to(u.AA, u.spectral())
+
+    @property
+    def energy(self):
+        return self._spectral_axis.to(u.ev, u.spectral())
+
+    @property
+    def velocity_convention(self):
+        return self._velocity_convention
+
+    @velocity_convention.setter
```

Comment:
however, having settable attributes like this gets back to the question of mutability.  In the `SpectralCube` implementation, we do `with_spectral_unit(u.km/u.s)` instead of `.velocity` as an attribute, since that avoids requiring settable `rest_value` and `velocity_convention`